src/main/java/bio/terra/service/filedata/azure/tables/TableDao.java

```diff
@@ -434,6 +437,9 @@ private FSItem makeFSFile(
     // pointing to the files in one or more datasets.
     FireStoreFile fireStoreFile =
         fileDao.retrieveFileMetadata(datasetTableServiceClient, datasetId, fileId);
+    if (fireStoreFile == null) {
+      throw new FileSystemExecutionException("Error retrieving file metadata for file");
```

Comment:
is this a separate issue? missing metadata compared to directory entries or is this the same?

Reply:
Yeah, exactly: Missing the file metadata (vs. directory entries).
This is just handling the case that the file metadata isn't returned for the given fileId. (which we're now allowing rather than throwing an exception)